examples/react/full-width-resizable-table/src/main.tsx

```diff
@@ -0,0 +1,158 @@
+            header: () => <span>Visits</span>,
+            footer: props => props.column.id,
+          }),
+          table.createDataColumn('status', {
+            header: 'Status',
+            footer: props => props.column.id,
+          }),
+          table.createDataColumn('progress', {
+            header: 'Profile Progress',
+            footer: props => props.column.id,
+          }),
+        ],
+      }),
+    ],
+    []
+  )
+
+  const [data, setData] = React.useState(() => makeData(20))
+  const refreshData = () => setData(() => makeData(20))
+
+  return (
+    <>
+      <Table
+        {...{
+          data,
+          columns,
+        }}
+      />
+      <hr />
+      <div>
+        <button onClick={() => rerender()}>Force Rerender</button>
+      </div>
+      <div>
+        <button onClick={() => refreshData()}>Refresh Data</button>
+      </div>
+    </>
+  )
+}
+
+function Table({
+  data,
+  columns,
+}: {
+  data: Person[]
+  columns: ColumnDef<typeof table.generics>[]
+}) {
+  const instance = useTableInstance(table, {
+    data,
+    columns,
+    enableColumnResizing: true,
+    columnResizeMode: 'onChange',
+
+    getCoreRowModel: getCoreRowModel(),
+    debugTable: true,
+  })
+
+  return (
+    <div className="p-2 block max-w-full overflow-x-scroll overflow-y-hidden">
+      <div className="h-2" />
+      <table className="w-full ">
+        <thead>
+          {instance.getHeaderGroups().map(headerGroup => (
+            <tr key={headerGroup.id}>
+              {headerGroup.headers.map(header => {
+                return (
+                  <th
+                    key={header.id}
+                    colSpan={header.colSpan}
+                    style={{ position: 'relative', width: header.getSize() }}
+                  >
+                    {header.isPlaceholder ? null : (
+                      <div>{header.renderHeader()}</div>
+                    )}
+                    {header.column.getCanResize() && (
+                      <div
+                        onMouseDown={header.getResizeHandler()}
+                        onTouchStart={header.getResizeHandler()}
+                        className={`resizer ${
+                          header.column.getIsResizing() ? 'isResizing' : ''
+                        }`}
+                      ></div>
+                    )}
+                  </th>
+                )
+              })}
+            </tr>
+          ))}
+        </thead>
+        <tbody>
+          {instance.getRowModel().rows.map(row => {
+            return (
+              <tr key={row.id}>
+                {row.getVisibleCells().map(cell => {
+                  return (
+                    <td key={cell.id} style={{ width: cell.column.getSize() }}>
+                      {cell.renderCell()}
+                    </td>
+                  )
+                })}
+              </tr>
+            )
+          })}
+        </tbody>
+      </table>
+    </div>
+  )
+}
+
+ReactDOM.render(
```

Comment:
can you upgrade to react 18.2.0
https://github.com/TanStack/table/blob/main/examples/react/basic/src/main.tsx#L169

Reply:
Example is now using React 18.2.0